app/forms/support_interface/application_forms/edit_reference_feedback_form.rb

```diff
@@ -0,0 +1,32 @@
+module SupportInterface
+  module ApplicationForms
+    class EditReferenceFeedbackForm
+      include ActiveModel::Model
+      include ValidationUtils
+
+      attr_accessor :feedback
+      attr_accessor :audit_comment
+
+      attr_reader :reference
+      attr_reader :application_form
+
+      validates :feedback, presence: true, word_count: { maximum: 500 }
+      validates :audit_comment, presence: true
+
+      def initialize(application_form, reference)
+        @application_form = application_form
+        @reference = reference
+
+        super(feedback: @reference.feedback)
+      end
+
+      def save!
+        @reference.update!(
+          feedback: feedback,
+          audit_comment: audit_comment,
+        )
+        SubmitReference.new(reference: @reference).save!
```

Comment:
By using this service it also triggers emails to the candidate and the referee and correctly changes the status of the reference

Reply:
It feels like we should enable some way to reverse this process too, i.e. if a support user leaves a blank feedback it sets the status to `feedback_requested`

Reply:
I spoke to @davidgisbey and we decided it best not to code for this for now. If we need to reverse the process this can still be done via the rails console.